frontend/rust-lib/flowy-document/src/event_handler.rs

```diff
@@ -391,3 +393,80 @@ pub(crate) async fn convert_data_to_json_handler(
 
   data_result_ok(ConvertDataToJsonResponsePB { json: result })
 }
+
+// Handler for uploading a file
+// `workspace_id` and `file_name` determines file identity
+pub(crate) async fn upload_file_handler(
+  params: AFPluginData<UploadFileParamsPB>,
+  manager: AFPluginState<Weak<DocumentManager>>,
+) -> DataResult<UploadedFile, FlowyError> {
+  let manager = upgrade_document(manager)?;
+  let file_service = manager
+    .get_file_storage_service()
+    .upgrade()
+    .ok_or(FlowyError::internal().with_context("The file storage service is already dropped"))?;
+  let url = file_service
+    .create_object(StorageObject {
+      workspace_id: params.workspace_id.clone(),
+      file_name: params.file_name.clone(),
+      value: ObjectValue::File {
+        file_path: params.local_file_path.clone(),
+      },
+    })
+    .await?;
+
+  let local_file_path = params.local_file_path.to_owned();
+  Ok(AFPluginData(UploadedFile {
+    url,
+    local_file_path,
+  }))
+}
+
+pub(crate) async fn download_file_handler(
+  params: AFPluginData<UploadedFile>,
```

Comment:
Download?

Reply:
This is a scenario where user signs into another device, loads a document and have to download file to his new device file system.

Reply:
No. I meant the type of the params. "download_file_handler" API with a "Upload" params.

Reply:
the params type is actually different from `upload_file_handler`. 
```
#[derive(Default, ProtoBuf)]
pub struct UploadedFilePB {
  #[pb(index = 1)]
  pub url: String,

  #[pb(index = 2)]
  pub local_file_path: String,
}
```
I read it as "downloading an uploaded file". I'm ok to change the names if it's confusing